### PR TITLE
Enable DDR tests for AIX

### PR DESCRIPTION
**Pull request by mikezhang1234567890**
Enables tests for DDR on AIX, closes https://github.com/eclipse/openj9/issues/1769
- enable callsitedbgddrext, classesdbgddrext, shrcdbgddrext, and
  modularityddrtests cmdlinetests for AIX
- enable DDR_Test for AIX
- create new AIX and zOS targets in playlist.xml for the DDR cmdline
  tests and use correct arguments to sysvcleanup.pl script. Pass in zos
  option since zos and aix options are identical in sysvcleanup.pl
- limit heap size in DDR tests to 256MB, since on AIX the core file
  scales with heap size, which can hit filesystem limits

Signed-off-by: Mike Zhang <mike.h.zhang@ibm.com>

This PR depends on the following extensions repo PRs, since the DDR tests do not compile without DDR enabled on AIX:
https://github.com/ibmruntimes/openj9-openjdk-jdk8/pull/116
https://github.com/ibmruntimes/openj9-openjdk-jdk9/pull/178
https://github.com/ibmruntimes/openj9-openjdk-jdk10/pull/59
https://github.com/ibmruntimes/openj9-openjdk-jdk11/pull/23
https://github.com/ibmruntimes/openj9-openjdk-jdk/pull/35

Doc issue is https://github.com/eclipse/openj9-docs/issues/92

### CODE REVIEW

File: test/functional/cmdLineTests/callsitedbgddrext/playlist.xml [hunk 1]
```diff
@@ -30,7 +30,7 @@
 		</variations>
 		<command>$(MKTREE) $(REPORTDIR); \
 	cd $(REPORTDIR); \
-	perl $(JVM_TEST_ROOT)$(D)TestConfig$(D)scripts$(D)tools$(D)sysvcleanup.pl zos ; \
+	perl $(JVM_TEST_ROOT)$(D)TestConfig$(D)scripts$(D)tools$(D)sysvcleanup.pl all zos ; \
```
Comment (pshipton) [inline test/functional/cmdLineTests/callsitedbgddrext/playlist.xml]: "all zos" is inconsistent with the later calls which specify "all"

File: test/functional/cmdLineTests/classesdbgddrext/playlist.xml [hunk 2]
```diff
@@ -24,13 +24,13 @@
...
 	<test>
-		<testCaseName>cmdLineTester_classesdbgddrext_zos</testCaseName>
+		<testCaseName>cmdLineTester_classesdbgddrext_aix_zos</testCaseName>
 		<variations>
 			<variation>NoOptions</variation>
 		</variations>
 		<command>$(MKTREE) $(REPORTDIR); \
 		cd $(REPORTDIR); \
-		perl $(JVM_TEST_ROOT)$(D)TestConfig$(D)scripts$(D)tools$(D)sysvcleanup.pl zos ; \
+		perl $(JVM_TEST_ROOT)$(D)TestConfig$(D)scripts$(D)tools$(D)sysvcleanup.pl all zos ; \
```
Comment (pshipton) [inline test/functional/cmdLineTests/classesdbgddrext/playlist.xml]: "all zos"?

File: test/functional/cmdLineTests/shrcdbgddrext/playlist.xml [hunk 3]
```diff
@@ -60,13 +59,13 @@
...
 	<test>
-		<testCaseName>cmdLineTester_shrcdbgddrext_zos</testCaseName>
+		<testCaseName>cmdLineTester_shrcdbgddrext_aix_zos</testCaseName>
 		<variations>
 			<variation>NoOptions</variation>
 		</variations>
 		<command>$(MKTREE) $(REPORTDIR); \
 	cd $(REPORTDIR); \
-	perl $(JVM_TEST_ROOT)$(D)TestConfig$(D)scripts$(D)tools$(D)sysvcleanup.pl zos ; \
+	perl $(JVM_TEST_ROOT)$(D)TestConfig$(D)scripts$(D)tools$(D)sysvcleanup.pl all zos ; \
```
Comment (pshipton) [inline test/functional/cmdLineTests/shrcdbgddrext/playlist.xml]: "all zos"?

File: test/functional/cmdLineTests/callsitedbgddrext/playlist.xml [hunk 4]
```diff
@@ -117,4 +116,39 @@
...
\ No newline at end of file
+
+	<test>
+		<testCaseName>cmdLineTester_callsitedbgddrext_openj9_aix_zos</testCaseName>
+		<variations>
+			<variation>NoOptions</variation>
+		</variations>
+		<command>$(MKTREE) $(REPORTDIR); \
+	cd $(REPORTDIR); \
+	perl $(JVM_TEST_ROOT)$(D)TestConfig$(D)scripts$(D)tools$(D)sysvcleanup.pl all zos ; \
```
Comment (pshipton) [inline test/functional/cmdLineTests/callsitedbgddrext/playlist.xml]: "all zos"?

File: test/functional/cmdLineTests/callsitedbgddrext/playlist.xml [hunk 5]
```diff
@@ -117,4 +116,39 @@
 			<subset>11</subset>
 		</subsets>
 	</test>
-</playlist>
\ No newline at end of file
+
+	<test>
```
Comment (pshipton) [inline test/functional/cmdLineTests/callsitedbgddrext/playlist.xml]: why is a specific test required for aix?

File: test/functional/cmdLineTests/callsitedbgddrext/playlist.xml [hunk 6]
```diff
@@ -91,15 +91,14 @@
 		</variations>
 		<command>$(MKTREE) $(REPORTDIR); \
 	cd $(REPORTDIR); \
-	perl $(JVM_TEST_ROOT)$(D)TestConfig$(D)scripts$(D)tools$(D)sysvcleanup.pl zos ; \
+	perl $(JVM_TEST_ROOT)$(D)TestConfig$(D)scripts$(D)tools$(D)sysvcleanup.pl all ; \
 	$(JAVA_COMMAND) $(JVM_OPTIONS) -Xint \
```
Comment (pshipton) [inline test/functional/cmdLineTests/callsitedbgddrext/playlist.xml]: You can remove the -Xint from all these tests. It's there for historical reasons when the jit was less reliable, but its no longer required. The tests will run faster.

File: test/functional/cmdLineTests/callsitedbgddrext/playlist.xml (hunk 1 above)
Comment (mikezhang1234567890) [reply test/functional/cmdLineTests/callsitedbgddrext/playlist.xml]: `all` is passed in to the `sysvcleanup.pl` script to ensure data from all users is cleaned up. AIX and ZOS have different user bits that are set by `sysvcleanup.pl` so the second option `zos` is passed when the test target is AIX/ZOS. `sysvcleanup.pl` accepts both `aix` and `zos` as a second option but they are identical.

**Comment by pshipton** [comment]
@mikezhang1234567890 I expect you've already noticed, but please fix the conflicts.

**Comment by mikezhang1234567890** [comment]
@keithc-ca or @pshipton can you review this please?

**Pull request merged**